src/launchCommand.ts

```diff
@@ -0,0 +1,68 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+import * as cp from "child_process";
+import * as vscode from "vscode";
+
+import { inferLaunchCommandLength } from "./languageServerPlugin";
+import { getJavaHome } from "./utility";
+
+enum shortenApproach {
+    none = "none",
+    jarmanifest = "jarmanifest",
+    argfile = "argfile",
+}
+
+export async function populateShortenCommandLineField(config: vscode.DebugConfiguration): Promise<vscode.DebugConfiguration> {
+    if (config.request === "launch" && (!config.shortenCommandLine || config.shortenCommandLine === "auto")) {
+        config.shortenCommandLine = await detectLaunchCommandStyle(config);
+    }
+
+    return config;
+}
+
+export async function detectLaunchCommandStyle(config: vscode.DebugConfiguration): Promise<shortenApproach> {
+    const cliLength = await inferLaunchCommandLength(config);
+    const javaHome = await getJavaHome();
+    const javaVersion = await checkJavaVersion(javaHome);
+    const recommendedApproach = javaVersion <= 8 ? shortenApproach.jarmanifest : shortenApproach.argfile;
+    if (config.console === "internalConsole") {

```

Comment:
If this command current for Windows, please add a logic such as os.platform == 'win32` to filter where this code applies